contracts/leaser/src/migrate.rs

```diff
@@ -2,101 +2,305 @@ use lease::api::MigrateMsg;
+    const CUSTOMER_ADDR2: &str = "customer2";
+    const CUSTOMER_ADDR3: &str = "customer3";
+    const CUSTOMER_ADDR4: &str = "customer4";
 
     #[test]
     fn no_leases() {
+        use std::array::IntoIter;
         let new_code = 242;
-        let no_leases = vec![];
+        let no_leases: Vec<Customer<IntoIter<Addr, 0>>> = vec![];
         assert_eq!(
-            Ok(Batch::default()),
-            super::migrate_leases(no_leases.into_iter().map(Ok), new_code)
+            Ok(MigrationResult::default()),
+            super::migrate_leases(no_leases.into_iter().map(Ok), 2, new_code)
+        );
+    }
+
+    #[test]
+    fn more_than_max_leases() {
+        let new_code: u64 = 242;
+        let lease1 = Addr::unchecked(LEASE1);
+        let lease2 = Addr::unchecked(LEASE21);
+        let lease3 = Addr::unchecked(LEASE22);
+        let customer_addr1 = Addr::unchecked(CUSTOMER_ADDR1);
+        let cust1 = Customer::from(
+            customer_addr1.clone(),
+            vec![lease1, lease2, lease3].into_iter(),
         );
+
+        let customers = vec![Ok(cust1)];
+        {
+            let exp = MigrationResult {
+                next_customer: Some(customer_addr1),
+                ..Default::default()
+            };
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(customers.into_iter(), new_code, 2)
+            );
+        }
+    }
+
+    #[test]
+    fn paging() {
+        let new_code = 242;
+        let lease1 = Addr::unchecked(LEASE1);
+        let lease21 = Addr::unchecked(LEASE21);
+        let lease22 = Addr::unchecked(LEASE22);
+        let lease3 = Addr::unchecked(LEASE3);
+        let lease41 = Addr::unchecked(LEASE41);
+        let lease42 = Addr::unchecked(LEASE42);
+        let lease43 = Addr::unchecked(LEASE43);
+        let customer_addr1 = Addr::unchecked(CUSTOMER_ADDR1);
+        let customer_addr2 = Addr::unchecked(CUSTOMER_ADDR2);
+        let customer_addr3 = Addr::unchecked(CUSTOMER_ADDR3);
+        let customer_addr4 = Addr::unchecked(CUSTOMER_ADDR4);
+
+        {
+            let exp = MigrationResult {
+                next_customer: Some(customer_addr1),
+                ..Default::default()
+            };
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(test_customers(), new_code, 0)
+            );
+        }
+        {
+            let mut exp = add_expected(MigrationResult::default(), &lease1, new_code);
+            exp.next_customer = Some(customer_addr2.clone());
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(test_customers(), new_code, 1)
+            );
+        }
+        {
+            let mut exp = add_expected(MigrationResult::default(), &lease1, new_code);
+            exp.next_customer = Some(customer_addr2);
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(test_customers(), new_code, 2)
+            );
```

Comment:
Expected expression is supposed to be the second argument to the macro.
```suggestion
            assert_eq!(
                super::migrate_leases(test_customers(), new_code, 2),
                Ok(exp)
            );
```

Reply:
same